src/omp-target/field_summary.cpp

```diff
@@ -82,9 +82,22 @@ void field_summary(global_variables &globals, parallel_ &parallel) {
     double *pressure = field.pressure.data;
     double *xvel0 = field.xvel0.data;
     double *yvel0 = field.yvel0.data;
+  
+    if (globals.profiler_on) {
+      globals.profiler.summary += timer() - kernel_time;
+      kernel_time = timer();
+    }
+
+#pragma omp target enter data map(to : vol) map(to : mass)   \
+    map(to : ie) map(to : ke) map(to : press)
+
+    if (globals.profiler_on) {
+      globals.profiler.host_to_device += timer() - kernel_time;
+      kernel_time = timer();
+    }
 
-#pragma omp target teams distribute parallel for simd clover_use_target(globals.context.use_target) map(tofrom : vol) map(tofrom : mass)   \
-    map(tofrom : ie) map(tofrom : ke) map(tofrom : press) reduction(+ : vol, mass, ie, ke, press)
+#pragma omp target teams distribute parallel for simd clover_use_target(globals.context.use_target) map(to : vol) map(to : mass)   \
+    map(to : ie) map(to : ke) map(to : press) reduction(+ : vol, mass, ie, ke, press)
```

Comment:
I think since we have all these map to clauses on the enter data directive we should delete them here to minimize ambiguity. Any decent compiler should know not to repeat the copy to the device but it's best imo to have them appear only once and in the place we expect them to happen for the timers.

Reply:
> I think since we have all these map to clauses on the enter data directive we should delete them here to minimize ambiguity. Any decent compiler should know not to repeat the copy to the device but it's best imo to have them appear only once and in the place we expect them to happen for the timers.

My understanding that not repeating the copies was part of the OpenMP standard, from looking at the documentation. In particular, it didn't seem like OpenMP had a separate directive equivalent to OpenACC's `present` which would explicitly state that the data is on the GPU other than the map directive

Reply:
Okay, we still pass the test cases when I delete the extra maps, so it's probably fine to do that

Reply:
Okay, pushed this change